shared/js/https.js

```diff
@@ -1,228 +1,211 @@
+                    load.JSONfromLocalFile(constants.httpsUpgradeList, (items) => {
+                        console.log("HTTPS: init() loaded upgrade list from local file with " + items.length + " items")
+                        httpsUpgradeList = items
+                        return resolve()
+                    })
+                })
             })
         })
     }
 
-    initSync () {
-        return new Promise((resolve) => {
-            this.db.logAllRecords(this.dbObjectStore, this.syncRuleCache)
-            resolve()
+    updateList() {
+        console.log("HTTPS: updateList() check if new list exists")
+
+        let etag = settings.getSetting('https-etag') || ''
+        
+        load.loadExtensionFile({
+            url: constants.httpsUpgradeListUrl,
+            source: 'external',
+            etag: etag
+        }, (data, res) => {
+            // this only gets called if the etag is different
+            // and it was able to get a new list
+            console.log("HTTPS: updateList() got updated list from server")
+
+            let newEtag = res.getResponseHeader('etag') || ''
+
+            try {
+                // update list in memory
+                httpsUpgradeList = JSON.parse(data)
+                console.log("HTTPS: updateList() new list has " + httpsUpgradeList.length + " items")
+                
+                this.saveToStorage(data)
+
+                // save new etag for next time
+                settings.updateSetting('https-etag', newEtag)
+                console.log("HTTPS: updateList() updated https-etag to " + newEtag)
+
+            } catch(e) {
+                console.log("HTTPS: updateList() error parsing server response")
+            }
         })
     }
 
-    ready () {
-        return this._ready
-    }
-
-    pipeRequestUrl (reqUrl, tab, isMainFrame) {
+    getFromStorage (fn) {
+        return new Promise((resolve, reject) => {
+            // For Chrome/Firefox:
+            if (window.chrome) {
+                console.log("HTTPS: getFromStorage() using chrome.storage.local (Chrome/FF)")
 
-        function checkDb (resolve) {
-            if (!this.isReady) {
-                console.warn('HTTPS: .pipeRequestUrl() this.db is not ready')
-                if (this.isSync) { return reqUrl } else { return resolve(reqUrl) }
-            }
+                chrome.storage.local.get('https-upgrade-list', (results) => {
+                    if (!results || !results['https-upgrade-list']) {
+                        return reject()
+                    }
 
-            // Only deal with http calls
-            const protocol = utils.getProtocol(reqUrl).toLowerCase()
-            if (!protocol.indexOf('http:') === 0) {
-                if (this.isSync) { return reqUrl } else { return resolve(reqUrl) }
-            }
+                    try {
+                        let parsedList = JSON.parse(results['https-upgrade-list'])
+                        resolve(parsedList)
```

Comment:
Can you explain why some of these have a `return` call and some don't? Obviously execution continues, but I'm not sure what happens after?

Reply:
I was only using returns when I needed to return early. Should be unnecessary here because it's either successfully finishing the try or falling to the catch and there's nothing afterwards.